### ISSUE TITLE
Cannot add red asterisk to a custom field on page

### ISSUE THREAD

**Issue by otherjoel** [hougaard/Simple-Object-Designer#187]
I’ve created a field named `vert_application`, and added it to the Sales Order page (among others).

![image](https://user-images.githubusercontent.com/1553115/231815669-10e25147-72f4-4b28-bc42-50c860005100.png)

This field happens to be an “Option” type field. 

Everything described so far is already published/deployed in production and working great.

Now I want to make this field mandatory. I can create a rule for it, but I can't add a red flag to it on the page because it doesn't appear in the dropdown:

![image](https://user-images.githubusercontent.com/1553115/231817135-687148c7-2d95-4b4c-abd1-b0226aa12213.png)

There are other fields on that page which have been added by other extensions which also do not appear in this dropdown.

Is it only possible to add red asterisks to fields which are built-in to the base BC?

**Comment by hougaard** [comment]
Hmm... Let me check that...

**Comment by otherjoel** [comment]
Just for clarification, was it determined to leave the behavior as-is for now? I’m still not able to add red asterisks to custom fields.

**Comment by hougaard** [comment]
It didn't appear after a symbol refresh?

**Comment by otherjoel** [comment]
No, I tried it just now. It’s the same situation — only “BC built in” fields appear in the “Red asterisk field (Main table)” dropdown. SOD prompts me to refresh symbols every single day as well, so I know it’s been done recently.

**Comment by hougaard** [comment]
But is the field in question visible on that page?

**Comment by otherjoel** [comment]
Oh sorry — yes, the field is definitely visible on the page. The mandatory condition-checking is working properly as well.

**Comment by hougaard** [comment]
Then my guess is, that it has a non-logical name...

**Comment by otherjoel** [comment]
What is meant by “non-logical”? 

As I say, other customizations in our environment (by other developers) have added other fields to that table (which are also visible on the Sales Order page), and none of them appear in this drop-down either.
